Source/VCGeneration/Split.cs

```diff
@@ -21,9 +20,50 @@ public record VCResult
     DateTime startTime,
     ProverInterface.Outcome outcome,
     TimeSpan runTime,
-    IEnumerable<AssertCmd> asserts,
+    int maxCounterExamples,
+    List<Counterexample> counterExamples,
+    List<AssertCmd> asserts,
     int resourceCount
-  );
+  ) {
+    public void ComputePerAssertOutcomes(out Dictionary<AssertCmd, ProverInterface.Outcome> perAssertOutcome,
+      out Dictionary<AssertCmd, Counterexample> perAssertCounterExamples) {
+      perAssertOutcome = new();
+      perAssertCounterExamples = new();
+      if (outcome == ProverInterface.Outcome.Valid) {
+        perAssertOutcome = asserts.ToDictionary(cmd => cmd, assertCmd => ProverInterface.Outcome.Valid);
+      } else {
+        foreach (var counterExample in counterExamples) {
+          // Only deal with the ocunter-examples that cover the asserts of this split.
```

Comment:
Don't all the counterexamples in `Split.CounterExamples` belong to this Split after your latest changes? Does the continue on line 49 ever trigger?

Reply:
Normally they should. If we were to write this in Dafny, I would just put some asserts. this is just a guard now in case something went wrong. I'll work on the comment.